pyiron/atomistics/master/elastic.py

```diff
@@ -0,0 +1,271 @@
+    return elastic_tensor_to_return
+
+def _fit_coeffs_with_stress(strain, stress, rotations):
+    stress = np.einsum('nik,mkl,njl->nmij', rotations, stress, rotations).reshape(-1, 3, 3)
+    stress = np.stack((stress[:,0,0], stress[:,1,1], stress[:,2,2], stress[:,1,2], stress[:,0,2], stress[:,0,1]), axis=-1)
+    score = []
+    coeff = []
+    for ii in range(6):
+        reg = LinearRegression().fit(strain, stress[:,ii])
+        coeff.append(reg.coef_)
+        score.append(reg.score(strain, stress[:,ii]))
+    return coeff, score
+
+def _fit_coeffs_with_energies(strain, energy, volume, rotations):
+    energy = np.tile(energy, len(rotations))
+    volume = np.tile(volume, len(rotations))
+    C = np.einsum('n,ni,nj->nij', 0.5*volume, strain, strain)
+    C = np.triu(C).reshape(-1, 36)
+    C = C[:,np.sum(C, axis=0)!=0]
+    C = np.append(np.ones(len(C)).reshape(-1, 1), C, axis=-1)
+    reg = LinearRegression().fit(C, energy)
+    score = reg.score(C, energy)
+    coeff = np.triu(np.ones((6,6))).flatten()
+    coeff[coeff!=0] *= reg.coef_[1:]*eV_div_A3_to_GPa
+    coeff = coeff.reshape(6,6)
+    coeff = 0.5*(coeff+coeff.T)
+    return coeff, score
+
+def calc_elastic_tensor(strain, stress=None, energy=None, volume=None, rotations=None, return_score=False):
+    """
+    Calculate 6x6-elastic tensor from the strain and stress or strain and energy+volume.
+
+    Rotations matrices can be added to take box symmetries into account (unit matrix can
+    be added but does not have to be included in the list)
+
+    Args:
+        strain (numpy.ndarray): nx3x3 strain tensors
+        stress (numpy.ndarray): nx3x3 stress tensors
+        energy (numpy.ndarray): n energy values
+        volume (numpy.ndarray): n volume values
+        rotations (numpy.ndarray): mx3x3 rotation matrices
+        return_score (numpy.ndarray): return regression score (cf. sklearn.linear_mode.LinearRegression)
+    """
+    if len(strain)==0:
+        raise ValueError('Not enough points')
+    if rotations is not None:
+        rotations = np.append(np.eye(3), rotations).reshape(-1, 3, 3)
+        _, indices = np.unique(np.round(rotations, 6), axis=0, return_inverse=True)
+        rotations = rotations[np.unique(indices)]
+    else:
+        rotations = np.eye(3)
+    strain = np.einsum('nik,mkl,njl->nmij', rotations, strain, rotations).reshape(-1, 3, 3)
+    strain = np.stack((strain[:,0,0], strain[:,1,1], strain[:,2,2], 2*strain[:,1,2], 2*strain[:,0,2], 2*strain[:,0,1]), axis=-1)
+    if stress is not None and len(stress)*len(rotations)==len(strain):
+        coeff, score = _fit_coeffs_with_stress(strain=strain, stress=stress, rotations=rotations)
+    elif energy is not None and volume is not None and len(energy)*len(rotations)==len(strain) and len(energy)==len(volume):
+        coeff, score = _fit_coeffs_with_energies(strain=strain, energy=energy, volume=volume, rotations=rotations)
+    else:
+        raise ValueError('Problem with fitting data')
```

Comment:
Message should be more specific, like "should have either stress or energy and volume".

Reply:
Yeah that's true. It already disturbed me during debugging :D